email_account_validity/_servlets.py

```diff
@@ -13,49 +13,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from twisted.web.resource import Resource
-
-from synapse.config._base import Config, ConfigError
-from synapse.http.server import (
+from synapse.module_api import (
     DirectServeHtmlResource,
     DirectServeJsonResource,
+    ModuleApi,
     respond_with_html,
 )
-from synapse.module_api import ModuleApi
-from synapse.module_api.errors import SynapseError
+from synapse.module_api.errors import ConfigError, SynapseError
 
 from email_account_validity._base import EmailAccountValidityBase
 from email_account_validity._store import EmailAccountValidityStore
 
 
-class EmailAccountValidityServlet(Resource):
-    def __init__(self, config: dict, api: ModuleApi):
-        super().__init__()
-        self.putChild(b'renew', EmailAccountValidityRenewServlet(config, api))
-        self.putChild(b'send_mail', EmailAccountValiditySendMailServlet(config, api))
-        self.putChild(b'admin', EmailAccountValidityAdminServlet(config, api))
-
-    @staticmethod
-    def parse_config(config: dict) -> dict:
-        config["period"] = Config.parse_duration(config.get("period") or 0)
-        return config
-
-
 class EmailAccountValidityRenewServlet(
     EmailAccountValidityBase, DirectServeHtmlResource
```

Comment:
having all the classes inherit from a base class which has a bunch of utility functions in is a pretty nasty antipattern. It's very easy to end up with too much coupling between the base class and the derived classes. It should be easy enough to move the utility functions elsewhere and pass in the required params. (see: https://en.wikipedia.org/wiki/Composition_over_inheritance, etc).

again, probably something to bear in mind for the future, rather than fix immediately.